controlpanel/frontend/consumers.py

```diff
@@ -126,6 +128,15 @@ def tool_deploy(self, message):
         else:
             log.debug(f"Deployed {tool.name} for {user}")
 
+    def tool_upgrade(self, message):
+        """
+        Upgrade simply means re-installing the helm chart for the tool
+        """
+        self.tool_deploy(message)
+
+        tool, user = self.get_tool_and_user(message)
+        update_tool_status(user, tool, TOOL_UPGRADED)
```

Comment:
I'm not keen on introducing new statuses - I don't think theres much value in differentiating between `Upgraded` and `Ready`.

Reply:
One good reason for being specific is that we know that we need to hide the "Upgrade" button once the tool is upgraded: Bear in mind we don't want to re-trigger a refresh and if we just recycle "ready" or "upgraded" we can't tell the difference.

Reply:
Ah, good point